readthedocs/search/documents.py

```diff
@@ -57,4 +56,59 @@ def faceted_search(cls, query, language=None, using=None, index=None):
 class PageDocument(DocType):
 
     class Meta(object):
-        model = HTMLFile
\ No newline at end of file
+        model = HTMLFile
+        fields = ('commit',)
+
+    project = fields.KeywordField(attr='project.slug')
+    version = fields.KeywordField(attr='version.slug')
+
+    title = fields.TextField(attr='processed_json.title')
+    headers = fields.TextField(attr='processed_json.headers')
+    content = fields.TextField(attr='processed_json.content')
+    path = fields.TextField(attr='processed_json.path')
+
+    @classmethod
+    def faceted_search(cls, query, projects=None, versions=None, using=None, index=None):
+        kwargs = {
+            'using': using or cls._doc_type.using,
+            'index': index or cls._doc_type.index,
+            'doc_types': [cls],
+            'model': cls._doc_type.model,
+            'query': query
+        }
+        filters = {}
+
+        if projects:
+            filters['project'] = projects
+        if versions:
+            filters['version'] = versions
+
+        kwargs['filters'] = filters
+
+        return FileSearch(**kwargs)
+
+    def get_queryset(self):
+        """Overwrite default queryset to filter certain files to index"""
+        queryset = super(PageDocument, self).get_queryset()
+
+        # Do not index files that belong to non sphinx project
+        # Also do not index certain files
+        queryset = (queryset.filter(project__documentation_type='sphinx')
+                            .exclude(name__in=SEARCH_EXCLUDED_FILE))
+        return queryset
+
+    def update(self, thing, **kwargs):
```

Comment:
When does this `update` actually get called? I believe it's a signal attached to the saving of the `ImportedFile` objects?

Reply:
Its called everytime the object is created, updated, or deleted.
Also whenever we run the management command. 
[call from the registry](https://github.com/sabricot/django-elasticsearch-dsl/blob/master/django_elasticsearch_dsl/registries.py#L75)